sys/riscv/riscv/locore.S

```diff
@@ -110,12 +111,53 @@ _start:
 	cjr	cra
 #else
 	j	mpentry
+#endif
+1:
+#ifdef __CHERI_PURE_CAPABILITY__
+	/* Store the boot hart */
+	cllc	ct0, boot_hart
+	csw	a0, 0(ct0)
+#else
+	/* Store the boot hart */
+	lla	t0, boot_hart
+	sw	a0, 0(t0)
+#endif
+	/* Load zero as modulep */
+	mv	a0, zero
+	j	pagetables
```

Comment:
Seems to work fine. It's in the same file so it's just an unconditional branch?

Reply:
`j sym` is `jal x0, sym`, so yes it does work, though at some point we'll want to make `jal` mode-dependent (`j` can still be permitted in assembly though, because whether it's `jal x0, sym` or `cjal c0, sym` doesn't change implicit register use in a meaningful way, unlike say `cret` that uses `cra` rather than `ra`).